docs/storage.md

```diff
@@ -20,4 +20,46 @@ NOTE: Currently Thanos requires strong consistency (write-read) for object store
 5. RUN the [TestObjStoreAcceptanceTest](/pkg/objstore/objtesting/acceptance_test.go) against your provider to ensure it fits. Fix any found error until test passes. (In PR)
 6. Add client implementation to the factory in [factory](/pkg/objstore/client/factory.go) code. (Using as small amount of flags as possible in every command)
 
-At that point, anyone can use your provider!
\ No newline at end of file
+At that point, anyone can use your provider!
+
+## S3 configuration
+
+Thanos uses minio client to upload Prometheus data into s3.
+
+To configure S3 bucket as an object store you need to set these mandatory S3 flags:
+- --s3.endpoint
+- --s3.bucket
+- --s3.access-key
+
+and set `S3_SECRET_KEY` environment variable with AWS secret key.
+
+Instead of using flags you can pass all the configuration via environment variables:
+- `S3_BUCKET`
+- `S3_ENDPOINT`
+- `S3_ACCESS_KEY`
+- `S3_SECRET_KEY`
+- `S3_INSECURE`
+- `S3_SIGNATURE_VERSION2`
+
+AWS region to endpoint mapping can be found in this [link](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region)
+
+If you are getting access denied errors try running with `--s3.signature-version2` or `--s3.insecure` flags.
```

Comment:
Not sure about this line, basically, you should use HTTPS all the time and use `insecure` only for debug purposes (also how using just HTTP fixes Access Denied error)?

Maybe:
```
Make sure you use a correct signature version with `--s3.signature-version2`, otherwise, you will get Access Denied error.
```

```
For debug purposes you can `--s3.insecure` to switch to plain insecure HTTP instead of HTTPS
```